pymc_bart/tree.py

```diff
@@ -231,9 +234,14 @@ def _traverse_tree(
         npt.NDArray[np.float_]
             Leaf node value or mean of leaf node values
         """
-        current_node: Node = self.get_node(node_index)
+        current_node = self.get_node(node_index)
+
         if current_node.is_leaf_node():
-            return np.array(current_node.value)
+            if current_node.linear_params is not None:
+                return np.array(current_node.value)
+            x = x[split_variable].item()
+            y_x = current_node.linear_params[0] + current_node.linear_params[1] * x
```

Comment:
In which method should we incorporate this logic? Also, where should we put the variables `m` and `split_variable` (what is the type of `split_variable`?)

Reply:
here https://github.com/pymc-devs/pymc-bart/blob/main/pymc_bart/tree.py#L259-L260

split_variable is of type integer the name of that variable in the new code is  `idx_split_variable`.

we should pass `m` to `predict` (predict is called from _sample_posterior which in turn is called from plot_variable_importance and plot_dependence ). Assuming  `μ` is the BART RV we can get `m` from `μ.owner.op.m`

Reply:
In https://github.com/pymc-devs/pymc-bart/pull/79/commits/8f5b293e7da723d831dc35c3dae7b8d063d07bd4, https://github.com/pymc-devs/pymc-bart/pull/79/commits/b31bbd0731339e92b10a4947b47fc15ff130f8cf and https://github.com/pymc-devs/pymc-bart/pull/79/commits/8cf70289dd874f3b61ddaff2897362f09a331fc6 I tried to add the signatures of the functions and the changes in the `tree.py` module. You suggested adding some changes to ` _traverse_leaf_values`, however as this function does not have an `x` argument, I thought adding the changes in `_traverse_tree` directly. I am of course open for feedback :) . For now, mypy and the tests are happy :)